### PR TITLE
allow recommend to come from server url.

### PR DESCRIPTION
**Pull request by vlandham**
results now store plot objects as they are easily serializable.

Changes work toward accomplishing #314 .

### CODE REVIEW

File: src/actions/result.ts [hunk 1]
```diff
@@ -18,22 +20,26 @@ export type ResultRequest = ReduxAction<typeof RESULT_REQUEST, {}>;
...
-export function resultRequest(query?: Query, schema?: Schema): AsyncResultRequest {
+export function resultRequest(query?: Query, schema?: Schema, data?: Data): AsyncResultRequest {
   return (dispatch: Dispatch<Action>, getState) => {
     if (!query) {
       query = getQuery(getState());
     }
     if (!schema) {
       schema = getSchema(getState());
     }
+    if (!data) {
```
Comment (kanitw) [inline src/actions/result.ts]: It seems like the input  `query`, `schema`, `data` are actually ignored and override

I wonder if we only need only one between (1) `query`, `schema`, `data` as parameters of `resultRequest`  or (2) `getState` as a parameter of the returned method.

Comment (vlandham) [inline src/actions/result.ts]: good point. I updated and removed them from the function signature. originally, the idea was to use this in multiple places - but based on how we are calling it now, it is not necessary. thanks!

**Comment by vlandham** [comment]
hmmm. the build process seems to be having problems installing yarn

```
Updating nvm
nvm.install
2.52s$ nvm install 6
Downloading and installing node v6.10.3...
Downloading https://nodejs.org/dist/v6.10.3/node-v6.10.3-linux-x64.tar.xz...
######################################################################## 100.0%
Computing checksum with sha256sum
Checksums matched!
Now using node v6.10.3 (npm v3.10.10)
Installing yarn
30.08s$ curl -o- -L https://yarnpkg.com/install.sh | bash

bash: line 1: syntax error near unexpected token `newline'
bash: line 1: `<!DOCTYPE html>'
The command "curl -o- -L https://yarnpkg.com/install.sh | bash" failed and exited with 2 during .
```

**Comment by vlandham** [comment]
getting a bad gateway error from https://yarnpkg.com/install.sh

**Comment by vlandham** [comment]
Looks like yarn fixed it self. 

@kanitw  - do you know if there is a way to rerun travis - or do i need to push an update to this branch?

**Comment by vlandham** [comment]
fixed @kanitw !

**Comment by kanitw** [comment]
Thanks.  Last comment, I'm wondering if we should add tests for `extractPlotObjects`, `convertToPlotObjectsGroup`?

**Comment by vlandham** [comment]
thanks for keeping me honest @kanitw ! Found an unnecessary input parameter to `extractPlotObjects` from the tests.

**Comment by vlandham** [comment]
now includes `buildSchema` in the api - so we can run that on the server as well.

**Comment by vlandham** [comment]
This tweak moves @types to dev dependencies - so they don't negatively impact `voyager-server` when it pulls in this repo.

**Comment by kanitw** [comment]
LGTM.  One last comment -- I wonder if it is difficult to add tests for `fetchCompassQLRecommend` and `fetchCompassQLBuildSchema`. 

I guess it will require some mock server?  If it doesn't take too long (say ~1hr), maybe it's worth adding a test for it?  If it's gonna way longer than that, let's merge this then.

**Comment by vlandham** [comment]
this latest push has a few tests for these two functions.

**Comment by kanitw** [comment]
LGTM.  It looks like there is a conflict with something I have just merged.  Do you wanna rebase to the latest vy2 branch?  After that we can merge :)

**Comment by vlandham** [comment]
ok ! i believe the conflicts have been resolved!

**Pull request merged**